src/main/java/com/cloudbees/jenkins/support/configfiles/SecretHandler.java

```diff
@@ -77,15 +81,37 @@ public void characters(char[] ch, int start, int length) throws SAXException {
                 super.characters(ch, start, length);
             }
         };
-        Source src = new SAXSource(xr, new InputSource(new StringReader(FileUtils.readFileToString(xmlFile))));
+        String str = FileUtils.readFileToString(xmlFile);
+        Source src = new SAXSource(xr, new InputSource(new StringReader(str)));
         final ByteArrayOutputStream result = new ByteArrayOutputStream();
         Result res = new StreamResult(result);
         Transformer transformer = TransformerFactory.newInstance().newTransformer();
         //omit xml declaration because of https://bugs.openjdk.java.net/browse/JDK-8035437
         transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
         transformer.setOutputProperty(OutputKeys.ENCODING, OUTPUT_ENCODING);
-        transformer.transform(src, res);
 
-        return result.toString("UTF-8");
+        try {
+            transformer.transform(src, res);
+            return result.toString("UTF-8");
+        } catch (TransformerException e) {
+            return findSecretFallback(str);
```

Comment:
Probably there should be fine-grain logging to notify about the issue

Reply:
I would also recommend this routine to be opt-out or opt-in. Skipping broken files is generally the most reasonable approach though you lose info

Reply:
I would vote for it. better safe than sorry